packages/admin/src/table/filterbar/filterBarPopoverFilter/FilterBarPopoverFilter.tsx

```diff
@@ -39,9 +41,10 @@ function PopoverFilter({
     return (
         <div className={classes.root}>
             <Form
-                onSubmit={(values) => {
-                    for (const name in values) {
-                        outerForm.change(name, values[name]);
+                onSubmit={(values, form) => {
+                    for (const name of form.getRegisteredFields()) {
+                        const outerFieldName = name.split(".")[0];
```

Comment:
Taking just the first part of the name seems wrong. What are you doing here?

Reply:
because thats always the key of the field name for the key value pairs of the outerForm Values.

Reply:
```form.getRegisteredFields()```  returns me this ```["user.firstname", "user.lastname"]``` so if i use it as you suggested i would get a values object like this: ```{"user.firstname": Maxi, "user.lastname": Frenslich}``` which is think not correct. I want an object like this: ```{user: {firstname: Maxi, lastname: Frenslich}}```. that was the main reason why i splitted the name and used only the first item of the array as the name. Correct me please if i am wrong but i almost sure its like that :)

And about the getIn Method, thanks for the advise :) its a better way to get the value of the object :)

Reply:
yes, the second form is the only correct one.

change() does support the dot-notation (as getIn does) I thought?

Reply:
i tried to do it but unfortunately it didnt work as expected. Otherwise i would have done it as you suggested :)

Reply:
what didn't work as expected? `change` or `getIn`?

Reply:
change didnt work like getIn :)

Reply:
I just tested this, this worked for me:
```
                    for (const name of form.getRegisteredFields()) {
                        const value = getIn(values, name);
                        outerForm.change(name, value);
                    }
```

what problem do you have with it?

Reply:
you are right ! it works, in the past i tried it also and it didnt work somehow :)